Dropdown menu for page navigation

DylanHNicolay opened the pull request:
# Description

This pull request implements a refactoring of HeaderButtonsComponents to use React Hooks instead of props. Additionally, it adds a dropdown menu to replace the buttons in the header. This change was made because the buttons were starting to wrap, causing the page layout to look worse.

Fixes #845 

## Type of change

(Check the ones that apply by placing an "x" instead of the space in the [ ] so it becomes [x])

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

(Note what you have done by placing an "x" instead of the space in the [ ] so it becomes [x]. It is hoped you do all of them.)

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [s] I have removed text in ( ) from the issue request

## Limitations

There is an issue where the help modal is appearing above the dropdown menu since using the css property, display:, will cause an error that prevents the menu from displaying properly. This issue could be related to the warning that pops up whenever the dropdown menu is clicked.

DylanHNicolay committed:
default

DylanHNicolay committed:
Current progress

DylanHNicolay committed:
Remove args from handleLogOut()

DylanHNicolay committed:
Convert HeaderButtonsComponent to utilize useSelector

DylanHNicolay committed:
Fix a bug where HeaderButtonsComponent was called using improper arguments

DylanHNicolay committed:
Implement placeholder for dropdown

DylanHNicolay committed:
Create temp dropdown menu

DylanHNicolay committed:
Fix a big where clicking the logout button (after signing in) does not log you out

DylanHNicolay committed:
Merge branch 'OpenEnergyDashboard:development' into temp

huss committed:
Get links working

- The links were not working due to an OED issue. To fix I used buttons
but this is not great. There is a TODO in the code with the issues.
- I changed the pages to use state and useEffect so it properly updates
on changes and initially.
- I added comments that should have been added long before these
changes began so not a comment on the current developer.
- MIsc. edits to clean up, mostly on old code before this effort.

huss committed:
Merge remote-tracking branch 'origin/temp' into temp

DylanHNicolay committed:
Resolve NPM errors

DylanHNicolay committed:
Added login and logout buttons to the dropdown menu

DylanHNicolay committed:
Merge branch 'OpenEnergyDashboard:development' into temp

DylanHNicolay committed:
Resolved an error with the dropdown menu and removed buttons

The Links contained inside Dropdown.Item caused an error to appear on the console.

Instead of styling the links as buttons, I used the as and to properties associated with Dropdown.Item
to achieve the better-looking dropdown menu without the error.

DylanHNicolay committed:
Removed unnecessary blank functions

DylanHNicolay committed:
Cleanup to comply with npm run check

huss commented on src/client/app/components/conversion/CreateConversionModalComponent.tsx:
```diff
@@ -273,4 +273,4 @@ export default function CreateConversionModalComponent(props: CreateConversionMo
 		</>
 	);
-}
+}
```
Please put the blank line back. Unsure why this file was edited.

DylanHNicolay committed:
add blank space

DylanHNicolay committed:
List in alpabetical order

DylanHNicolay committed:
Added relevant TODO

DylanHNicolay committed:
Removed trailing spaces to comply with npm run check

DylanHNicolay committed:
Fix a bug where clicking on the menu button would cause a warning to appear

DylanHNicolay committed:
Add a TODO item for a bug

huss committed:
fix small formatting issues

huss committed:
added comments on tooltip issue

huss committed:
try to fix lint error only on GitHub

huss reviewed (commented):
Thanks to @Nicolax0 for this PR. I appreciate your efforts and success in getting the links to show as desired and getting rid of the console messages. This PR works as desired except for one remaining item which is the help tooltip does not display as desired. I have tried to fix this but have yet to succeed. I'm going to try again later but this PR will need to be merged either way.

huss committed:
fix help issue & translate

The issue with the help not working is fixed where a comment is left
about why it was happening. The fixed allowed the help icon to be
placed below the menu as desired. The menu title is now translated.

huss reviewed (approved):
I fixed the last issue and this now works as desired. It is ready to merge.
